spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/MappingSpannerReadConverter.java

```diff
@@ -115,44 +66,61 @@
 	public <R> R read(Class<R> type, Struct source, Set<String> includeColumns,
 			boolean allowMissingColumns) {
 		boolean readAllColumns = includeColumns == null;
-		R object = instantiate(type);
 		SpannerPersistentEntity<?> persistentEntity = this.spannerMappingContext
 				.getPersistentEntity(type);
 
-		PersistentPropertyAccessor accessor = persistentEntity
-				.getPropertyAccessor(object);
+		StructAccessor structAccessor = new StructAccessor(source);
+
+		StructPropertyValueProvider propertyValueProvider = new StructPropertyValueProvider(structAccessor, this);
+
+		PreferredConstructor<?, SpannerPersistentProperty> persistenceConstructor = persistentEntity
+				.getPersistenceConstructor();
+
+		// @formatter:off
+		ParameterValueProvider<SpannerPersistentProperty> parameterValueProvider =
+						new PersistentEntityParameterValueProvider<>(persistentEntity, propertyValueProvider, null);
+		// @formatter:on
+
+		EntityInstantiator instantiator = this.instantiators.getInstantiatorFor(persistentEntity);
+		Object instance = instantiator.createInstance(persistentEntity, parameterValueProvider);
+		PersistentPropertyAccessor accessor = persistentEntity.getPropertyAccessor(instance);
 
 		persistentEntity.doWithProperties(
 				(PropertyHandler<SpannerPersistentProperty>) spannerPersistentProperty -> {
-					if (!shouldSkipProperty(source,
+					if (!shouldSkipProperty(
```

Comment:
This `shouldSkipProperty` function is now effectively just the few function calls with ORs among them. I think you should just do those OR checks inside this if-statement here. 

For somebody not familiar with the code this `shouldSkipProperty` function looks like a mysterious function that takes half a dozen very specific and uncommon arguments with obvious connections to each other.

Reply:
This is a simple readability method. 
It is simpler, shorter. 
It means that there is a method that has the property skipping logic, that depends on all of these things. If you're interested, have a look at it. 
I tend to use these as I believe that they result in [cleaner code](https://medium.com/coding-skills/clean-code-101-meaningful-names-and-functions-bf450456d90c).